services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/Persistence.kt

```diff
@@ -80,7 +80,7 @@ enum class TyomaaTyyppi {
 }
 
 /** NOTE Järjestys täytyy olla pienimmästä suurimpaan */
-enum class TodennakoinenHaittaPaaAjoRatojenKaistajarjestelyihin(
+enum class VaikutusAutoliikenteenKaistamaariin(
```

Comment:
The enums here could be moved under the tormaystarkastelu package, maybe in `TormaystarkasteluLuokittelu.kt` with the other ones.

Reply:
Moved enums to Enums.kt files in root hanke package and tormaystarkastelu package. This way there are no references from upper packages to lower packages.

Reply:
Renaming `TormaystarkasteluLuokittelu.kt` to less descriptive `Enums` doesn't make much sense to me.

Why shouldn't there be references from parent packages to child packages?

Reply:
For me it makes sense and is comparable to Persistence.kt or Domain.kt or Constants.kt and such.

Referencing in my opinion should always go from botton to top in order to avoid circular dependencies. Perhaps this idea is outdated, I don't know.

Reply:
I think generic names like that should only be used if a more descriptive name can't be found.

Circular dependencies happen with injecting dependencies, not with class references, if I'm not mistaken.